### ISSUE TITLE
False positive from "prefer-function-over-method" when only using class's generic type

### ISSUE THREAD

**Issue by erkalicious** [palantir/tslint#3090]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: Atom

#### TypeScript code being linted

```ts
class Class<T> {
	example() {
		// Do something meaningful with T
		return (item: T) => true 
	}
}
```

with `tslint.json` configuration:

```json
"prefer-function-over-method": true
```

#### Actual behavior
Error on method `example`: Class method does not use 'this'. Use a function instead. (prefer-function-over-method)

#### Expected behavior
No error here, as `example` must be a method in the class to immediately know about the type T

**Comment by adidahiya** [comment]
This would work and is equally type-safe:

```ts
function example<T>() {
    return (item: T) => true;
}
```

I don't think this warrants a change in rule behavior.

**Comment by erkalicious** [comment]
Yes, but that would also require the programmer to specificy the generic type each function call, whereas the method call automatically knows the generic type of the instance on which it is called.

**Comment by adidahiya** [comment]
> that would also require the programmer to specificy the generic type each function call

That doesn't sound right; it should be inferred by the compiler in most cases.

**Comment by erkalicious** [comment]
@adidahiya yes, I stand corrected. I don't have a case where the compiler doesn't infer the type off the top of my head. If I come across one, I'll update this thread. We can close it out for now. Thanks!
